### ISSUE TITLE
Is there a hooks could create actor ?

### ISSUE THREAD

**Issue by himyyy** [Connect2IC/connect2ic#8]

**Comment by MioQuispe** [comment]
Do you mean canister?
There is:
https://connect2ic.github.io/docs/docs/React/hooks#usecanister

It abstracts all details away of how the canister is created, since every wallet does it differently.

**Comment by himyyy** [comment]
> Do you mean canister? There is: https://connect2ic.github.io/docs/docs/React/hooks#usecanister
> 
> It abstracts all details away of how the canister is created, since every wallet does it differently.

Using the canister method should knowing canisterId, but the canisterId may be unknown, maybe store on ic.Therefore, I think export the createActor method just like Plug extension is a better way.

**Comment by MioQuispe** [comment]
> > Do you mean canister? There is: https://connect2ic.github.io/docs/docs/React/hooks#usecanister
> > It abstracts all details away of how the canister is created, since every wallet does it differently.
> 
> Using the canister method should knowing canisterId, but the canisterId may be unknown, maybe store on ic.Therefore, I think export the createActor method just like Plug extension is a better way.

So you want to use mainnet canisters is that correct? I have an issue open for that https://github.com/Connect2IC/connect2ic/issues/5 

It seems like something worth supporting for sure. And yeah would also like to make it able to handle dynamic canisterIds where its not known beforehand. Will have to think about it however what the API should look like. Im not sure exposing the createActor is the way to go.

**Comment by himyyy** [comment]
Yeah, i want to handle the dynamic canisterIds. Thanks for your support, please let me know when you finished it.

**Comment by MioQuispe** [comment]
> Yeah, i want to handle the dynamic canisterIds. Thanks for your support, please let me know when you finished it.

In the meantime, you could do:

```js
const {activeProvider} = useConnect()

activeProvider.connector.createActor(canisterId, idlFactory)
```

**Comment by himyyy** [comment]
> > Yeah, i want to handle the dynamic canisterIds. Thanks for your support, please let me know when you finished it.
> 
> In the meantime, you could do:
> 
> ```js
> const {activeProvider} = useConnect()
> 
> const actor = await activeProvider.connector.createActor(canisterId, idlFactory)
> ```
Thanks.

**Comment by himyyy** [comment]
> > Yeah, i want to handle the dynamic canisterIds. Thanks for your support, please let me know when you finished it.
> 
> In the meantime, you could do:
> 
> ```js
> const {activeProvider} = useConnect()
> 
> const actor = await activeProvider.connector.createActor(canisterId, idlFactory)
> ```
Sorry for reopen it.
Is there any way to get this actor by function method, not hooks

**Comment by neeboo** [comment]
> > > Yeah, i want to handle the dynamic canisterIds. Thanks for your support, please let me know when you finished it.
> > 
> > 
> > In the meantime, you could do:
> > ```js
> > const {activeProvider} = useConnect()
> > 
> > const actor = await activeProvider.connector.createActor(canisterId, idlFactory)
> > ```
> 
> Sorry for reopen it. Is there any way to get this actor by function method, not hooks

I  think the best way to use them as function method meanwhile using custom canister, is to wrap the function with connector.

```typescript

async function createActorFromConnector<T>(connector, canisterId, idlFactory):Promise<ActorSubClass<T>>{
       // ...
}

and return Actor somehow


```

**Comment by himyyy** [comment]
> 
For now, all i know how to get the "connector" is use the useConnect hooks.But it's seems to be hard for me to use hooks, because of the older code's logic

**Comment by neeboo** [comment]
> For now, all i know how to get the "connector" is use the useConnect hooks.But it's seems to be hard for me to use hooks, because of the older code's logic

We can also consider wrapping window API for future version.

**Comment by MioQuispe** [comment]
> > 
> 
> For now, all i know how to get the "connector" is use the useConnect hooks.But it's seems to be hard for me to use hooks, because of the older code's logic

Yes I will probably expose the "client" in the future so it can be used outside

**Comment by MioQuispe** [comment]
> > 
> 
> For now, all i know how to get the "connector" is use the useConnect hooks.But it's seems to be hard for me to use hooks, because of the older code's logic

Hey there, check out latest version 0.1.0. The client is now exposed
So there's no need to use hooks https://connect2ic.github.io/docs/docs/Core/Client
